cai_causal_graph/time_series_causal_graph.py

```diff
@@ -811,19 +811,49 @@ def add_edge(
         if edge is not None:
             assert isinstance(edge, self._EdgeCls), f'Edge class must be of type {self._EdgeCls.__name__}.'
             # Check source node type
-            source_node = edge.source
+            source_node = deepcopy(edge.source)
             if isinstance(source_node, Node) and not isinstance(source_node, self._NodeCls):
                 source_node = self._NodeCls.from_dict(source_node.to_dict())
             # Check destination node type
-            destination_node = edge.destination
+            destination_node = deepcopy(edge.destination)
             if isinstance(destination_node, Node) and not isinstance(destination_node, self._NodeCls):
                 destination_node = self._NodeCls.from_dict(destination_node.to_dict())
             # See if either node was recreated, if yes, then create new edge with appropriate nodes
             if source_node != edge.source or destination_node != edge.destination:
+                if edge_type != EdgeType.DIRECTED_EDGE:
+                    # if time_lag for source is greater than destination, then swap source and destination
+                    assert isinstance(source, TimeSeriesNode) and isinstance(destination, TimeSeriesNode)
+                    if source_node.time_lag > destination_node.time_lag:
+                        destination_node, source_node = source_node, destination_node
                 edge = self._EdgeCls(
                     source_node, destination_node, edge_type=edge.get_edge_type(), meta=edge.get_metadata()
                 )
 
+        else:
```

Comment:
Can you just add a comment here so user knows what is happening. Lots of new code and essentially just putting the nodes in the right format. But just explain this else block a bit.